uniflow/model/model.py

```diff
@@ -33,40 +33,39 @@ def __init__(
         """
         model_server_cls = ModelServerFactory.get(model_config["model_server"])
         self._model_server = model_server_cls(model_config)
-        if isinstance(guided_prompt_template, GuidedPrompt):
-            self._guided_prompt_template = guided_prompt_template.get_prompt()
-        else:
-            self._guided_prompt_template = guided_prompt_template
+        self._guided_prompt_template = guided_prompt_template
         self._prompt_keys = []
 
     def _serialize(self, data: List[Context]) -> List[str]:
         """Serialize data.
 
         Args:
-            data (List[Dict[str, Any]]): Data to serialize.
+            data (List[Context]): Data to serialize.
 
         Returns:
             List[str]: Serialized data.
         """
         output = []
         for d in data:
+            if not isinstance(d, Context):
+                if "context" in d:
+                    d = Context(context=d["context"])
+                else:
+                    raise ValueError(
+                        "Input data must be a Context object or have a 'context' field."
+                    )
+            output_strings = []
             guided_prompt_template = copy.deepcopy(self._guided_prompt_template)
-            if isinstance(d, Context):
-                d = d.model_dump()
-            if "examples" in guided_prompt_template:
-                guided_prompt_template["examples"].append(d)
+            if isinstance(guided_prompt_template, GuidedPrompt):
+                guided_prompt_template.examples.append(d)
+                output_strings.append(
+                    f"instruction: {guided_prompt_template.instruction}"
+                )
+                for example in guided_prompt_template.examples:
+                    for ex_key, ex_value in example.model_dump().items():
+                        output_strings.append(f"{ex_key}: {ex_value}")
             else:
-                guided_prompt_template = d
-
-            output_strings = []
-            # Iterate over each key-value pair in the dictionary
-            for key, value in guided_prompt_template.items():
-                if isinstance(value, list):
-                    # Special handling for the "examples" list
-                    for example in value:
-                        for ex_key, ex_value in example.items():
-                            output_strings.append(f"{ex_key}: {ex_value}")
-                else:
+                for key, value in d.model_dump().items():
```

Comment:
I am very confused on the logic here for the else case that you are not using `GuidedPrompt` then?